### ISSUE TITLE
PEP 696 false positive when used with a class with no `__new__`

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#4417]
When a generic class uses default TypeVar types and provides an `__init__` but no `__new__` method, pyright computes an incorrect specialization type for the constructor call.

```python
from dataclasses import dataclass
from typing import TypeAlias


class Commentable: ...

class User(Commentable): ...
class ClientUser(Commentable): ...

Authors: TypeAlias = User | ClientUser
@dataclass
class Comment[OwnerT: Commentable = Commentable, AuthorT: Authors = Authors]:
    author: AuthorT
    owner: OwnerT

our_user = ClientUser()

def post_comment[OwnerT: Commentable](owner: OwnerT) -> Comment[OwnerT, ClientUser]:
    return Comment(author=our_user, owner=owner)
```

**Comment by erictraut** [comment]
@Gobot1234, this will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.288, which I just published. It will also be included in a future release of pylance.
